### ISSUE TITLE
[BUG] Siptrace using HEP over UDP exhausts package memory when DNS cannot be resolved

### ISSUE THREAD

**Issue by bcnewlin** [OpenSIPS/opensips#1782]
**OpenSIPS version you are running**

```
version: opensips 2.4.6 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 4e258b17f
main.c compiled on 18:41:52 Jul 26 2019 with gcc 7
```

**Describe the bug**

We had an issue where the DNS record for our sipcapture server was mistakenly deleted. When this happened, OpenSIPS was unable to send HEP traces to the capture server. We found that after being in this state for some time, OpenSIPS began reporting E_CORE_PKG_THRESHOLD errors. We believe the siptrace module was not properly releasing memory used to store the messages due to the DNS error and the module ran out of memory.

This was accompanied by several crashes, I do not know whether they were due to the memory exhaustion or another cause. I have opened a separate ticket for the crashes.

**To Reproduce**

1. Start OpenSIPS, configured with HEP over UDP using an unresolvable DNS name
2. Start traffic
3. Watch package memory for siptrace module should increase consistently

**Expected behavior**

OpenSIPS should release memory associated with a traced message if it fails to send the trace.

**Relevant System Logs**

```
CRITICAL:core:mk_proxy: could not resolve hostname: "capture-server.com"
ERROR:proto_hep:send_hep_message: bad hep host name!
ERROR:siptrace:save_siptrace: Failed to duplicate with hep to <capture-server.com>
```

**OS/environment information**
 - Operating System:  Amazon Linux 2
 - OpenSIPS installation:  git
 - other relevant information:

**Additional context**

**Comment by bogdan-iancu** [comment]
@bcnewlin , it will be great if you could confirm the fix

**Comment by bcnewlin** [comment]
@bogdan-iancu Thank you! We are still blocked on upgrading to the latest version due to #1771 but as soon as that is resolved I can test.

**Comment by bcnewlin** [comment]
We were able to test and confirm this fix.
